doc/rst/language/spec/statements.rst

```diff
@@ -533,6 +553,21 @@ termination expression.
       4
       5
 
+The control-flow declaration *ctrl-decl*, when used in a while-do loop,
+works similarly to how it does in a conditional statement.
+It declares a variable whose scope is the loop body.
+Its *expression* must be of a class type and is considered to evaluate
+to ``true`` when its value is not ``nil``.
+If the expression's type is ``borrowed`` or  ``unmanaged``,
+the variable being declared stores the expression's value and
+the variable's type is the non-nilable variant (:ref:`Nilable_Classes`)
+of the expression's type.  Otherwise the variable stores a borrow of
+the expression's value (:ref:`Class_Lifetime_and_Borrows`) and has the
+non-nillable ``borrowed`` counterpart of the expression's type.  The
+variable can be modified within the loop body if it is declared with
+the ``var`` keyword.  The expression may be evaluated repeatedly as if
+it were used as the while-expression without the declaration.
```

Comment:
I would rephrase this, since the expression itself is not what evaluates to `true` or `false`.  Maybe "as if it were compared to ``nil`` in the while-expression"?